src/AppBundle/Controller/Report/ReportController.php

```diff
@@ -443,6 +445,37 @@ public function unsubmit(Request $request, $id)
         return ['id' => $report->getId()];
     }
 
+    /**
+     * Update users's reports cached status when not set
+     * Flushes every 5 records to allow resuming in case of timeouts
+     *
+     * @param $userId
+     */
+    private function updateReportStatusCache($userId)
+    {
+        $em = $this->get('em');
+        $repo = $em->getRepository(Report::class);
+
+        while (($reports = $repo
+                ->createQueryBuilder('r')
+                ->select('r,c,u')
+                ->leftJoin('r.client', 'c')
+                ->leftJoin('c.users', 'u')
+                ->where('u.id = :uid')
+                ->andWhere('r.reportStatusCached IS NULL')
+                ->setParameter('uid', $userId)
+                ->setMaxResults(5)
+                ->getQuery()
+                ->getResult()) && count($reports)
+        ) {
```

Comment:
This line will be executed on every run through of the loop. Needs moving outside the while statement. Do query once, and use results in while loop

Reply:
could you put some comment above this to explain what its updating please?

Reply:
Building queries belongs int the repository class to allow reuse across other controllers

Reply:
I need to execute at each loop, as each loop flushes 5 records (and help at next reload in case of probable crash at first load). This code can be removed when all the reports have a cache, so I didn't put in a repo for reuse